src/nexgen/nxs_utils/Source.py

```diff
@@ -0,0 +1,88 @@
+"""
+Object definition for Source, Beam and Attenuator
+"""
+from __future__ import annotations
+
+from dataclasses import dataclass
+
+from dataclasses_json import dataclass_json
+
+from ..utils import Facility
+
+
+class Source:
+    def __init__(
+        self,
+        beamline: str,
+        facility: Facility = Facility(
+            "Diamond Light Source",
+            "DLS",
+            "Synchrotron X-ray Source",
+            None,
+        ),
+        probe: str | None = None,
+    ):
+        self.beamline = beamline
+
+        self.name = facility.name
+        self.short_name = facility.short_name
+        self.facility_type = facility.type
+        self.facility_id = facility.id
+
+        self.probe = probe
+
+    def __repr__(self) -> str:
+        msg = f"Beamline {self.beamline} at {self.name}.\n"
+        if self.probe:
+            msg += "Probe: {}"
+        return msg
+
+    def set_instrument_name(self) -> str:
```

Comment:
This is never called in production and is a bit confusingly named given it doesn't set anything, just returns you what it calculates.